### PR TITLE
docs: fix cpu.shares part in man pages

### PR DESCRIPTION
**Pull request by hqhq**
The original description has some mistakes and lack of many useful
information, I rewrite them to make it accurate and complete.

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

**Comment by SvenDowideit** [comment]
LGTM - @fredlf @jamtur01 @spf13 @moxiegirl

File: docs/man/docker-run.1.md [hunk 1]
```diff
@@ -82,24 +82,35 @@ option can be set multiple times.
...
-with time quantum set as 100 milliseconds, containers C0, C1, and C2 will run
-for full-time quantum, and container C3 will run for half-time quantum i.e 50
-milliseconds.
+   You can modify the priority of a container
+with the `-c` option. By default, all containers run at the same priority
+and get the same proportion of CPU cycles, the value specified must be 2
+or higher, if you are not setting `-c` or `--cpu-shares`, the default
+shares of CPU time would be 1024.
+
+CPU shares is kind of CPU bandwidth weight, the proportion will only
```
Comment (jamtur01) [inline docs/man/docker-run.1.md]: Sentence structure?

File: docs/man/docker-run.1.md [hunk 2]
```diff
@@ -82,24 +82,35 @@ option can be set multiple times.
...
+reflect when CPU-intensive processes are running. When tasks in one
+container are idle, other containers are allowed to borrow the left-over
+CPU time.
+
+The actual amount of CPU time can very depending on the number of containers
+running on the system. If a container have a share of 1024 and two other
```
Comment (jamtur01) [inline docs/man/docker-run.1.md]: has a share

File: docs/man/docker-run.1.md [hunk 3]
```diff
@@ -82,24 +82,35 @@ option can be set multiple times.
...
+containers have share of 512, when processes in all containers attempt to
```
Comment (jamtur01) [inline docs/man/docker-run.1.md]: have shares of ...

File: docs/man/docker-run.1.md [hunk 4]
```diff
@@ -82,24 +82,35 @@ option can be set multiple times.
...
+use 100% of CPU, the first container would receive 50% of all CPU time, if
+another container with share of 1024 is added, the first container would
```
Comment (jamtur01) [inline docs/man/docker-run.1.md]: a share

File: docs/man/docker-run.1.md [hunk 5]
```diff
@@ -82,24 +82,35 @@ option can be set multiple times.
...
+only get 33% of the CPU (the rest receive 16.5%, 16.5% and 33% of CPU).
+
+Note that shares of CPU time are distributed per all CPU cores on multi-core
+systems. Even if a container is limited to less than 100% of CPU time, it
+may use 100% of each individual CPU core. E.g., if we start {C0} container
+with (`-c` = 512) and {C1} with (`-c` = 1024), we start three CPU-intensive
+processes (one in {C0} and two in {C1}) on a system with more than three
+cores, might results in the following division of CPU shares:
```
Comment (jamtur01) [inline docs/man/docker-run.1.md]: sentence doesn't make sense.

File: docs/sources/reference/run.md [hunk 6]
```diff
@@ -329,24 +329,35 @@ We have four ways to set memory usage:
...
-proportion of CPU cycles.
-
-E.g., If we start three {C0, C1, C2} containers with default values
-(`-c` OR `--cpu-shares` = 0) and one {C3} with (`-c` or `--cpu-shares`=512)
-then C0, C1, and C2 would have access to 100% CPU shares (1024) and C3 would
-only have access to 50% CPU shares (512). In the context of a time-sliced OS
-with time quantum set as 100 milliseconds, containers C0, C1, and C2 will run
-for full-time quantum, and container C3 will run for half-time quantum i.e 50
-milliseconds.
+The operator can modify the priority of this container with
```
Comment (jamtur01) [inline docs/sources/reference/run.md]: Edits as above.

**Comment by moxiegirl** [comment]
Try this for your first file, I hope I captured your itention:
-----
By default, all containers run at the same priority. This means they all get the
same proportion of CPU cycles (1024). Use the `-c` or `--cpu-shares` option to
modify a container's priority. You must set the priority to 2 or higher. 

CPU shares is kind of CPU bandwidth weight. The proportion of cycles reflect the
number of CPU-intensive processes. When tasks in one container are idle, other
containers can borrow the left-over CPU time.The actual amount of CPU time can
vary depending on the number of containers running. 

For example, consider three containers, one container has a share of 1024 and
two other containers have 512 each. When processes on the three containers
attempt to use 100% of CPU, the first container receives 50% of all CPU time. If
you add a fourth container with a 1024 share, the first container would only get
33% of the CPU. The remaining containers receive 16.5%, 16.5% and 33% of CPU.

On a multi-core system, the shares of CPU time are distributed per all CPU
cores. Even if a container is limited to less than 100% of CPU time, it can use
100% of each individual CPU core. 

For example, consider a system with more than three cores. If you start {C0}
container with (`-c` = 512) , {C1} with (`-c` = 1024), and three CPU-intensive
processes (one in {C0} and two in {C1}) this can results in the following
division of CPU shares:

PID	container	CPU	CPU share
100	{C0}		0	100% of CPU0
101	{C1}		1	100% of CPU1
102	{C1}		2	100% of CPU2

**Comment by SvenDowideit** [comment]
Hey @hqhq thankyou lots for this, you've added quite a bit of correctness.

I'm going to take over this PR and change some of the wording and then see where things go :)

**Comment by hqhq** [comment]
@SvenDowideit sure, if you want to change context more than words, you can ping me to check the correctness.

**Comment by SvenDowideit** [comment]
replaced by #11146

**Pull request closed without merging**